spirv_glsl.cpp

```diff
@@ -3395,6 +3395,33 @@ void CompilerGLSL::emit_resources()
 		statement("");
 
 	declare_undefined_values();
+
+	if (requires_transpose_2x2)
```

Comment:
This is a very strange to place this. Do this in emit_extension_workarounds perhaps. That's where most "wrapper" code is.

Reply:
I put it here because the HLSL implementations writes the helper functions in the exact same location.  I can move it though.